users/views.py

```diff
@@ -19,3 +19,17 @@ def update(self, *args, **kwargs):
         serializer.is_valid(raise_exception=True)
         serializer.save()
         return Response(serializer.data, status=200)
+
+
+class LoginAPI(ViewSet):
+    """ Login viewset. 
+    """
+    def post(self, *args, **kwargs):
+        serializer = LoginSerializer(data=self.request.data)
+
+        if serializer.is_valid():
+            user = User.objects.get(email=serializer.data.get('email'))
+            token = Token.objects.create(user=user)
```

Comment:
do not do this here. avoid adding logic to the view. instead place it in the serializer or model. 

http://www.django-rest-framework.org/api-guide/authentication/#generating-tokens